pypdf/_cmap.py

```diff
@@ -423,57 +434,96 @@ def compute_space_width(
             st = w[0] if isinstance(w[0], int) else w[0].get_object()
             second = w[1].get_object()
             if isinstance(second, int):
-                for x in range(st, second):
-                    w1[x] = w[2]
+                # C_first C_last same_W
+                en = second
+                for c_code in range(st, en + 1):
+                    try:
+                        conversion_char = map_dict[chr(c_code)]
+                        font_width_map[ord(conversion_char)] = w[2]
+                    except Exception:
+                        pass
                 w = w[3:]
             elif isinstance(second, list):
-                for y in second:
-                    w1[st] = y
-                    st += 1
+                # Starting_C [W1 W2 ... Wn]
+                c_code = st
+                for width in second:
+                    try:
+                        conversion_char = map_dict[chr(c_code)]
+                        font_width_map[ord(conversion_char)] = width
+                    except Exception:
+                        pass
+                    c_code += 1
                 w = w[2:]
             else:
                 logger_warning(
                     "unknown widths : \n" + (ft1["/W"]).__repr__(),
                     __name__,
                 )
                 break
-        try:
-            sp_width = w1[space_code]
-        except Exception:
+    elif "/Widths" in ft:
+        w = list(ft["/Widths"])
```

Comment:
Accessing an entry of a `DictionaryObject` will yield a `PdfObject`, which *mypy* complains about here. The old code just ignored it.

The best way would be to check that the value is the expected `ArrayObject` you probably want here. An `ArrayObject` extends the list class, thus calling `list` on it should work now - although I am not sure whether we actually need this as an `ArrayObject` is a list already.

Reply:
@stefan6419846
Thank you! Code style checks passed.